src/Illuminate/Database/Eloquent/Relations/HasOne.php

```diff
@@ -77,12 +76,22 @@ public function match(array $models, Collection $results, $relation)
      */
     protected function getDefaultFor(Model $model)
     {
+        $result = null;
+
         if (is_callable($this->withDefault)) {
-            return call_user_func($this->withDefault);
+            $result = call_user_func($this->withDefault);
```

Comment:
1. I don't know, I just wanted to leave that possibility.
2. I really like how you refactored the code 👍